Client/Migration/BraveCoreMigrator.swift

```diff
@@ -128,36 +128,53 @@ public class BraveCoreMigrator {
       return
     }
 
+    let group = DispatchGroup()
+    
+    group.enter()
+    
     // Step 1:  Check If bookmarks are migrated / migrate
     migrateBookmarkModels { [unowned self] success in
       guard success else {
         self.migrationObserver = .failed
         completion?(.failedBookmarksMigration)
-
+        
         return
       }
+      
+      group.leave()
+    }
+    
+    group.enter()
+    
+    // Step 2: Check If history is migrated / migrate
+    migrateHistoryModels { [unowned self] success in
+      guard success else {
+        self.migrationObserver = .failed
+        completion?(.failedHistoryMigration)
+        
+        return
+      }
+      
+      group.leave()
```

Comment:
This isn't called if you return early (due to error). Ensure you place it in a defer instead, i.e. `defer { group.leave() }`

Reply:
Fix at https://github.com/brave/brave-ios/pull/5641/commits/e745114a3e090226f302b99b880622defa67ea46